File: Decorator/Python/Person.py
class Person:
    def get_information(self):
        pass

class Student(Person):
    def __init__(self, name, high_school_year):
        self.name = name
        self.high_school_year = high_school_year

    def get_information(self):
        return f"Name: {self.name}, High School Year: {self.high_school_year}"

class ProfessionDecorator(Person):
    def __init__(self, person, profession):
        self.person = person
        self.profession = profession

    def get_information(self):
        return f"{self.person.get_information()}, Profession: {self.profession}"

class Workplace(Person):
    def __init__(self, person, workplace):
        self.persona = person
        self.workplace = workplace

    def get_information(self):
        return f"{self.person.get_information()}, Workplace: {self.workplace}"

if __name__ == "__main__":
    student = Student("Juan Perez", 2020)

    student = ProfessionDecorator(student, "Engineer")

    student = Workplace(student, "Empress ABC")

    print(student.get_information())

File: Decorator/JavaScript/Person.js
class Person {
    getInformation() {}
}

class Student extends Person {
    constructor(name, highSchoolYear) {
        super();
        this.name = name;
        this.highSchoolYear = highSchoolYear;
    }

    getInformation() {
        return `Name: ${this.name}, High School Year: ${this.highSchoolYear}`;
    }
}

class ProfessionDecorator extends Person {
    constructor(person, profession) {
        super();
        this.person = person;
        this.profession = profession;
    }

    getInformation() {
        return `${this.person.getInformation()}, Profession: ${this.profession}`;
    }
}

class Workplace extends Person {
    constructor(person, workplace) {
        super();
        this.person = person;
        this.workplace = workplace;
    }

    getInformation() {
        return `${this.person.getInformation()}, Workplace: ${this.workplace}`;
    }
}

const student = new Student("Juan Perez", 2020);
const studentProffesion = new ProfessionDecorator(student, "Engineer");
const completeStudent = new Workplace(studentProffesion, "Empress ABC");

console.log(completeStudent.getInformation());
